packages/dialog/src/vaadin-dialog.js

```diff
@@ -144,15 +134,46 @@ class Dialog extends DialogSizeMixin(
         @mousedown="${this._bringOverlayToFront}"
         @touchstart="${this._bringOverlayToFront}"
         theme="${ifDefined(this._theme)}"
-        aria-label="${ifDefined(this.ariaLabel || this.headerTitle)}"
         .modeless="${this.modeless}"
         .withBackdrop="${!this.modeless}"
         ?resizable="${this.resizable}"
         restore-focus-on-close
         focus-trap
-      ></vaadin-dialog-overlay>
+        exportparts="backdrop, overlay, header, title, header-content, content, footer"
+      >
+        <slot name="title" slot="title"></slot>
+        <slot name="header-content" slot="header-content"></slot>
+        <slot name="footer" slot="footer"></slot>
+        <slot></slot>
+      </vaadin-dialog-overlay>
     `;
   }
+
+  /** @protected */
+  updated(props) {
+    super.updated(props);
+
+    if (props.has('headerTitle')) {
+      this.__applyHeaderTitleAriaLabel();
+    }
+  }
+
+  /** @private */
+  __applyHeaderTitleAriaLabel() {
+    // If there is a header title and no aria-label,
+    // set the aria-label to the header title.
+    if (this.headerTitle && !this.ariaLabel) {
+      this.ariaLabel = this.headerTitle;
+      this.__useHeaderTitleAriaLabel = true;
+    }
+
+    // If there is no header title and the aria-label was set
+    // using a header title, remove the aria-label.
+    if (this.__useHeaderTitleAriaLabel && !this.headerTitle) {
+      this.ariaLabel = null;
+      this.__useHeaderTitleAriaLabel = false;
+    }
+  }
```

Comment:
Likewise, this is supposed to allow using `aria-label` instead of `headerTitle`. 

Put this into dialog itself, as `DialogBaseMixin` is used in CRUD where we need different logic and it doesn't have `headerTitle`.

Reply:
>  If updated doesn't include headerTitle if it's never set

Yes, I think that's the case. 

Technically it's possible that `ariaLabel` is set, then `headerTitle` is set, and then `ariaLabel` gets removed - in theory, we probably should then set default `ariaLabel` based on `headerTitle`. But it's probably an edge case.